src/Servers/IIS/IIS/src/Core/Native/SecPkgContext_CipherInfo.cs

```diff
@@ -0,0 +1,30 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.InteropServices;
+
+namespace Microsoft.AspNetCore.Server.IIS.Core.Native;
+
+// From Schannel.h
+[StructLayout(LayoutKind.Sequential, CharSet = CharSet.Unicode)]
+internal unsafe struct SecPkgContext_CipherInfo
```

Comment:
From an engineering/quality/maintability perspective, I'd really like to see the two definitions of this type in this repo be merged down into one.  Does cswin32 not define it?

I don't see an obvious place that the IIS and Http.Sys modules can share code, so maybe this is as good as it gets.  But two copies of the struct, two copies of calling HttpQueryRequestProperty (which don't even do it the same way, even though they could)... strikes me as "not great".

Reply:
I am also not happy that cswin32 is not updated properly and does not provide the types itself - i created an issue to follow-up on that.
https://github.com/dotnet/aspnetcore/issues/63686

I agree duplication of types is not nice, but as well dont see an intersecting place to locate this type unfortunately :(

Reply:
Oh cool, thanks for finding that. I put the `SecPkgContext_CipherInfo` under `Shared\Httpsys\NativeInterop`.